lib/chef/provider/aws_vpc.rb

```diff
@@ -37,45 +32,20 @@ class Chef::Provider::AwsVpc < Chef::Provisioning::AWSDriver::AWSProvider
     end
   end
 
-  action :delete do
-    aws_object = new_resource.aws_object
-    if aws_object
-      ig = aws_object.internet_gateway
-      if ig
-        converge_by "detach Internet Gateway #{ig.id} in #{region} from VPC #{new_resource.name} (#{aws_object.id}" do
-          ig.detach(aws_object.id)
-        end
-        if ig.tags['OwnedByVPC'] == aws_object.id
-          converge_by "destroy Internet Gateway #{ig.id} in #{region} (owned by VPC #{new_resource.name} (#{aws_object.id}))" do
-            ig.delete
-          end
-        end
-      end
-      # TODO delete main route table & routes if they exist
-      # TODO (?) detach dhcp options
-      converge_by "delete VPC #{aws_object.id} in #{region}" do
-        aws_object.delete
-      end
-    end
+  protected
 
-    new_resource.delete_managed_entry(action_handler)
-  end
+  def create_aws_object
+    options = { }
+    options[:instance_tenancy] = new_resource.instance_tenancy if new_resource.instance_tenancy
 
-  private
-
-  def create_vpc
-    vpc = nil
     converge_by "create new VPC #{new_resource.name} in #{region}" do
-      opts = { }
-      opts[:instance_tenancy] = new_resource.instance_tenancy if new_resource.instance_tenancy
-      vpc = driver.ec2.vpcs.create(new_resource.cidr_block, opts)
+      vpc = new_resource.driver.ec2.vpcs.create(new_resource.cidr_block, options)
       vpc.tags['Name'] = new_resource.name
```

Comment:
Future note: All the tag-able AWS objects include the module [AWS::EC2::TaggedItem](http://docs.aws.amazon.com/AWSRubySDK/latest/AWS/EC2/TaggedItem.html)